File: codeforces/90.collecting_coins_[1294A]/collecting_coins.cpp
#include <iostream>
#include <algorithm>
/* Author: José Rodolfo (jric2002) */
using namespace std;
int main() {
  unsigned short int t;
  int a, b, c, n, m, r;
  cin >> t;
  while (t--) {
    cin >> a >> b >> c >> n;
    m = max(max(a, b), c);
    r = n - ((m - a) + (m - b) + (m - c));
    if (((r % 3) == 0) && (r >= 0)) {
      cout << "YES" << endl;
    }
    else {
      cout << "NO" << endl;
    }
  }
  return 0;
}

File: codeforces/90.collecting_coins_[1294A]/collecting_coins.py
# Author: José Rodolfo (jric2002)
t = int(input())
while (t):
  a, b, c, n = map(int, input().split(" "))
  m = max(a, b, c)
  r = n - ((m - a) + (m - b) + (m - c))
  print("YES") if (((r % 3) == 0 and (r >= 0))) else print("NO")
  t -= 1